packages/core/src/config.ts

```diff
@@ -405,7 +410,7 @@ async function composeLibRsbuildConfig(
     config.bundle,
     dirname(configPath),
   );
-  const dtsConfig = await composeDtsConfig(config);
+  const dtsConfig = await composeDtsConfig(config, dtsExtension);
```

Comment:
Could we move the dts logic in `composeAutoExtensionConfig` into `composeDtsConfig`, then DTS config will not couple with JS anymore.

Reply:
I aim to reduce multiple time call for `getDefaultExtension`
![image](https://github.com/user-attachments/assets/25ecb8e3-aef8-4fea-a80f-8d7f795ac2b6)
 Or is this acceptable?

Reply:
So be it like this, we could do some refactor in the future.